src/main/java/org/example/page/vietjet/HomePage.java

```diff
@@ -0,0 +1,145 @@
+    }
+
+    @Step("Select data: {0}")
+    public void selectDatePicker(LocalDate localDate) {
+        String country = (String) YamlUtils.getProperty("config.country");
+        String language = (String) YamlUtils.getProperty("config.language");
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("MMMM yyyy", new Locale.Builder()
+                .setLanguage(language)
+                .setRegion(country)
+                .build());
+        getDate(localDate, formatter).click();
+    }
+
+    @Step("Select number of adults: {0}")
+    public void selectNumberOfAdults(int targetNumber) {
+        int currentNumber = Integer.parseInt(adultsValue.getText());
+        SelenideElement buttonToClick = currentNumber < targetNumber ? plusAdultsButton : minusAdultsButton;
+        for (int i = 0; i < Math.abs(targetNumber - currentNumber); i++) {
+            buttonToClick.click();
+        }
+    }
+
+    @Step("Select number of adults: {0}")
+    public void selectNumberOfChildren(int targetNumber) {
+        int currentNumber = Integer.parseInt(childrenValue.getText());
+        SelenideElement buttonToClick = currentNumber < targetNumber ? plusChildrenButton : minusChildrenButton;
+        for (int i = 0; i < Math.abs(targetNumber - currentNumber); i++) {
+            buttonToClick.click();
+        }
+    }
+
+    @Step("Select number of adults: {0}")
+    public void selectNumberOfInfants(int targetNumber) {
+        int currentNumber = Integer.parseInt(babyValue.getText());
+        SelenideElement buttonToClick = currentNumber < targetNumber ? plusBabyButton : minusBabyButton;
+        for (int i = 0; i < Math.abs(targetNumber - currentNumber); i++) {
+            buttonToClick.click();
+        }
+    }
+
+    @Step("Click on Search button")
+    public void clickSearchButton() {
+        getButtonByText((String) YamlUtils.getProperty("button.letgo")).click();
+    }
+
+    private SelenideElement getTextboxByLabel(Label label) {
+        return $x(String.format("//div[label[.='%s']]//input", label.toString()))
+                .shouldBe(Condition.visible);
+    }
+
+    private SelenideElement getDate(LocalDate localDate, DateTimeFormatter formatter) {
+        String monthYear = localDate.format(formatter);
+        return $x(String.format(date, monthYear, localDate.getDayOfMonth())).shouldBe(Condition.visible);
+    }
+
+    private SelenideElement getButtonByText(String text) {
+        return $x(String.format(button, text)).shouldBe(Condition.visible);
+    }
+
+    private SelenideElement acceptCookiesButton = $x("//button[contains(@class,'MuiButton-root')]//h5");
+    private SelenideElement iframe = $("#preview-notification-frame");
+    private SelenideElement notNowButton = $("#NC_CTA_TWO");
+    private SelenideElement roundTripRadio = $("input[value='roundTrip']");
+    private SelenideElement adultsValue = $x("//div[div[img[contains(@alt,'adults')]]]/following-sibling::div/span");
```

Comment:
Try this(removed the 2nd [ ] pair and replace them by /). It is less ambiguous, easier to maintain, and still precise.

```suggestion
    private SelenideElement adultsValue = $x("//div[div/img[contains(@alt,'adults')]]/following-sibling::div/span");
```

Reply:
updated